### PR TITLE
Replace exact numDocs by soft-del count in SegmentCommitInfo

### PR DESCRIPTION
**Pull request by dnhatn**
This PR adapts/utilizes recent enhancements in Lucene-7.4:

1. Replaces exactNumDocs by the soft-deletes count in SegmentCommitInfo. This enhancement allows us to back out changes introduced in #30228.

2.  Always configure the soft-deletes field in IWC

### CODE REVIEW

**Comment by dnhatn** [comment]
run sample packaging tests

**Comment by s1monw** [comment]
@elasticmachine run sample packaging tests

File: server/src/test/java/org/elasticsearch/index/query/MatchPhrasePrefixQueryBuilderTests.java [hunk 1]
```diff
@@ -39,6 +40,7 @@
 import static org.hamcrest.Matchers.containsString;
 import static org.hamcrest.Matchers.notNullValue;
 
+@LuceneTestCase.AwaitsFix(bugUrl = "should-be-fixed-by-upstream")
```
Comment (s1monw) [inline server/src/test/java/org/elasticsearch/index/query/MatchPhrasePrefixQueryBuilderTests.java]: can you elaborate what caused this?

**Review by s1monw — APPROVED** [review]
LGTM I am glad we dont' have to jump through hoops anymore for the doc stats.

**Review by bleskes — APPROVED** [review]
LGTM. Nice stats :)

File: server/src/test/java/org/elasticsearch/index/query/MatchPhrasePrefixQueryBuilderTests.java (hunk 1 above)
Comment (dnhatn) [reply server/src/test/java/org/elasticsearch/index/query/MatchPhrasePrefixQueryBuilderTests.java]: These tests are failing on the master branch.

**Comment by dnhatn** [comment]
Thanks @s1monw and @bleskes

**Pull request merged**